db/models/user.test.js

```diff
@@ -0,0 +1,147 @@
+
+  let user
+
+  beforeEach(function(){
+    user = User.build({
+      first_name: 'Simon',
+      last_name: 'Cat',
+      user_name: 'simoncat',
+      email: 'scat@gmail.com',
+      password: 'sssccc',
+      is_admin: false
+    })
+  })
+
+  describe('includes all correct attributes,', () => {
+    it('has correct values', () => {
+      return user.save()
+      .then( (savedUser) => {
+        expect(savedUser.first_name).to.equal('Simon');
+        expect(savedUser.last_name).to.equal('Cat');
+        expect(savedUser.user_name).to.equal(simoncat);
+        expect(savedUser.email).to.equal('scat@gmail.com');
+        expect(savedUser.password).to.equal('sssccc');
+        expect(savedUser.is_admin).to.equal(false);
+      })
+    });
+  })
+
+  describe('Validations ', () => {
+    it('errors when no first name is entered', function () {
+      user.first_name = null
+
+      return user.validate()
+      .then(function (result) {
+        expect(result).to.be.an.instanceOf(Error);
+        expect(result.message).to.contain('Validation error')
+      })
+    })
+
+     it('requires `first name` to not be an empty string', function () {
+
+      user.first_name = '';
+
+      return user.validate()
+      .then(function (result) {
+        expect(result).to.be.an.instanceOf(Error);
+        expect(result.message).to.contain('Validation error')
+      })
+    })
+
+     it('errors when no last name is entered', function () {
+      user.last_name = null
+
+      return user.validate()
+      .then(function (result) {
+        expect(result).to.be.an.instanceOf(Error);
+        expect(result.message).to.contain('Validation error')
+      })
+    })
+
+     it('requires `last name` to not be an empty string', function () {
+
+      user.last_name = '';
+
+      return user.validate()
+      .then(function (result) {
+        expect(result).to.be.an.instanceOf(Error);
+        expect(result.message).to.contain('Validation error')
+      })
+    })
+
+
+    it('errors when no user name is entered', function () {
+      user.user_name = null
+
+      return user.validate()
+      .then(function (result) {
+        expect(result).to.be.an.instanceOf(Error);
+        expect(result.message).to.contain('Validation error')
+      });
+    });
+
+    it('errors when no email is entered', function () {
+      user.email = null
+
+      return user.validate()
+      .then(function (result) {
+        expect(result).to.be.an.instanceOf(Error);
+        expect(result.message).to.contain('Validation error')
+      });
+    });
+
+    it('errors when  email format is invalid', function () {
+      user.email = 'clearly_not_an_email'
+
+      return user.validate()
+      .then(function (result) {
+        expect(result).to.be.an.instanceOf(Error);
+        expect(result.message).to.contain('Validation error')
+      });
+    });
+
+
+    it('errors when no password is entered' , function () {
+      user.password = null
```

Comment:
What other validations do we want on passwords?  Can we make tests for length for example?  Or if it doesn't contain a symbol?

  it('errors when password is too short' , function () {
      user.password = 'abcde'

      return user.validate()
      .then(function (result) {
        expect(result).to.be.an.instanceOf(Error);
        expect(result.message).to.contain('Validation error')
      });
    });